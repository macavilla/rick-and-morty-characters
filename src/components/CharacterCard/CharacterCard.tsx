import Image from "next/image";

import { CharacterType, CharacterStatus } from "@/types/character";

export interface CharacerCardProps {
  character: CharacterType;
  onSelect: () => void;
  isSelected: boolean;
  isDisabled?: boolean;
}

const CharacterCard = ({
  character,
  onSelect,
  isSelected,
  isDisabled,
}: CharacerCardProps) => {
  const handleClick = () => {
    if (!isDisabled) onSelect();
  };

  return (
    <div
      onClick={handleClick}
      className={`
      p-2 border-2 rounded-3xl lg:flex lg:gap-3 
      ${
        isDisabled
          ? "cursor-not-allowed opacity-50"
          : "cursor-pointer hover:text-gray-500"
      }
      ${isSelected && "border-2 border-indigo-950 text-indigo-700 font-bold"}
    `}
    >
      <div className="w-full lg:w-1/2">
        <div className="relative">
          <Image
            src={character.image}
            alt={character.name}
            width={300}
            height={300}
            className="w-full h-auto  rounded-2xl"
          />
          <div className="flex items-center gap-2 absolute top-1 left-1 border-1 bg-foreground border-background text-background rounded-full py-0.5 px-1.5">
            <div
              aria-hidden={true}
              className={`rounded-full size-3  ${
                (character.status === CharacterStatus.alive &&
                  "bg-green-500") ||
                (character.status === CharacterStatus.dead && "bg-red-500") ||
                "bg-gray-500"
              }`}
            ></div>
            <p className="capitalize text-xs">{character.status}</p>
          </div>
        </div>
      </div>
      <div className="lg:w-1/2 py-2  mt-2 lg:m-0 lg:p-0  lg:pl-3">
        <h4 className="text-xl lg:text-2xl mb-2">{character.name}</h4>
        <p className="italic text-lg">{character.species}</p>
      </div>
    </div>
  );
};

export default CharacterCard;
