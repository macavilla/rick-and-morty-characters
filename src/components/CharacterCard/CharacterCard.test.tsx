import { screen, render } from "@testing-library/react";
import {
  mockCharacter,
  mockCharacterCardProps,
} from "./__mockData__/CharacterCard.mock";

import CharacterCard from "./CharacterCard";

describe("CharacterCard", () => {
  it("renders CharacterCard and matches snapshot", () => {
    const { container } = render(<CharacterCard {...mockCharacterCardProps} />);
    expect(container).toMatchSnapshot();
  });

  // it("should render character data", () => {
  //   render(<CharacterCard {...mockCharacterCardProps} />);

  //   const characterName = screen.getByText(mockCharacter.name);
  //   expect(characterName).toBeInTheDocument();
  // });
});
