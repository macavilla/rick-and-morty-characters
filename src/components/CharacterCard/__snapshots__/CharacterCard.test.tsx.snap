// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CharacterCard renders CharacterCard and matches snapshot 1`] = `
<div>
  <div
    class="
      p-2 border rounded-md lg:flex lg:gap-1
      cursor-pointer hover:text-gray-500
      false
    "
  >
    <div
      class="w-full lg:w-1/2"
    >
      <img
        alt="Rick Sanchez"
        class="w-full h-auto"
        data-nimg="1"
        decoding="async"
        height="300"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
    <div
      class="lg:w-1/2"
    >
      <h4
        class="text-2xl"
      >
        Rick Sanchez
      </h4>
      <div
        class="flex items-center gap-1"
      >
        <div
          class="rounded-full size-3 mb-0.5 bg-green-500"
        />
        <p
          class="capitalize"
        >
          Alive
        </p>
      </div>
      <p>
        Human
      </p>
    </div>
  </div>
</div>
`;
